src/decode.rs

```diff
@@ -4674,23 +4674,20 @@ fn get_upscale_x0(in_w: c_int, out_w: c_int, step: c_int) -> c_int {
 
 pub unsafe fn rav1d_submit_frame(c: &mut Rav1dContext) -> Rav1dResult {
     // wait for c->out_delayed[next] and move into c->out if visible
-    let (fc, out, _task_thread_lock) = if c.n_fc > 1 {
+    let (fc, out, _task_thread_lock) = if c.fc.len() > 1 {
         let mut task_thread_lock = c.task_thread.delayed_fg.lock().unwrap();
         let next = c.frame_thread.next;
-        c.frame_thread.next += 1;
-        if c.frame_thread.next == c.n_fc {
-            c.frame_thread.next = 0;
-        }
+        c.frame_thread.next = (c.frame_thread.next + 1).wrapping_rem(c.fc.len() as u32);
```

Comment:
```suggestion
        c.frame_thread.next = (c.frame_thread.next + 1) % c.fc.len() as u32;
```

`.wrapping_rem` is a no-op.